hgshah/cmput404-project
# Liking a Post and Comment

Changes
- Like a post or comment locally or remotely
- You can only like a post or comment once!
- This effectively got rid of post, comments and follow :(
- It doesn't do any checking to see that the person we're sending the like to is the author in the object id lol ... maybe something I'll do later

Inbox of Local Follower
![Screen Shot 2022-11-18 at 11 38 16 PM](https://user-images.githubusercontent.com/56659167/202838653-2ca6306e-fd24-4164-bc85-7c6020662867.png)

Inbox of Remote Follower
![Screen Shot 2022-11-18 at 11 38 22 PM](https://user-images.githubusercontent.com/56659167/202838671-3d6209d2-5420-41de-a9f3-e83bf5fdc7bc.png)

Like Post of Remote Follower
![Screen Shot 2022-11-18 at 11 38 46 PM](https://user-images.githubusercontent.com/56659167/202838690-e9344fa2-3a0b-40ce-bdc6-a305c33b46ca.png)

Like Comment of Local Follower
![Screen Shot 2022-11-18 at 11 39 25 PM](https://user-images.githubusercontent.com/56659167/202838697-fead19c1-978c-4904-8ff0-efe0b79ba3d9.png)

Inbox of Local Follower
![Screen Shot 2022-11-18 at 11 39 40 PM](https://user-images.githubusercontent.com/56659167/202838709-cd493d5f-a89a-4777-b04e-282ec854e526.png)

Inbox of Remote Follower
![Screen Shot 2022-11-18 at 11 39 51 PM](https://user-images.githubusercontent.com/56659167/202838721-559e593a-9ac1-4a08-817a-e917ce91e1f6.png)

How to test:
1. Have two of our servers running: port 8000 and 8080
2. Create two authors (amanda, monica) on ServerA and one on ServerB (chris)
4. Create a Post for monica on Server A. Create a Comment on a Post for chris on Server B (this author will need to make their own post, no remote functionality to comment on a post on another server)
5. Follow the screenshots above to see the endpoints author amanda needs to call to like a post and comment
6. To see the inbox of monica and chris, make sure you're logged in as them and do: `author/AUTHOR_ID/inbox/all`
